tests/finetune/eval/if_eval/test_rule_factory.py

```diff
@@ -57,6 +64,36 @@ def test_ends_with_has_no_commas(self):
             prompt = rule.get_prompt()
             self.assertTrue("," not in prompt)
 
+    def test_generate_if_eval_sample_v1_excludes_v2(self):
+        dummy_qa = ("", "")
+
+        for _ in range(1000):
+            sample = generate_if_eval_sample(dummy_qa, dummy_qa, 1, 1, IfEvalVersion.V1)
+            rule_id_set = set([rule.rule_id for rule in sample.rules])
+            self.assertTrue(rule_id_set.isdisjoint(V2_RULES))
+
+    def test_generate_if_eval_sample_v2_includes_v1(self):
+        dummy_qa = ("", "")
+
+        included_v1 = False
+        included_v2 = False
+
+        joint_rules = V1_RULES | V2_RULES
+
+        for _ in range(1000):
+            sample = generate_if_eval_sample(dummy_qa, dummy_qa, 1, 1, IfEvalVersion.V2)
+            rule_id_set = set([rule.rule_id for rule in sample.rules])
+            self.assertTrue(rule_id_set.issubset(joint_rules))
+            # Check if we used a v1 rule at least once.
+            if not included_v1:
+                included_v1 = not rule_id_set.isdisjoint(V1_RULES)
+            # Check if we used a v2 rule at least ocne.
```

Comment:
I guess we should also check that none of them are DummyRule?

Reply:
Added